server/stt/stt_planning_ml.py

```diff
@@ -124,7 +136,221 @@ async def get_coverage_details(
             # Return ``None`` so this coverage isn't added to the Planning item
             return None
 
-        return await super().get_coverage_details(news_coverage_elt, item, original)
+        coverage = await super().get_coverage_details(news_coverage_elt, item, original)
+        if coverage is not None:
+            # Only parse STT-specific fields for new coverages or when we have the specific STT data
+            self.parse_stt_coverage_fields(news_coverage_elt, coverage)
+        return coverage
+
+    def parse_stt_coverage_fields(
+        self, news_coverage_elt: Element, coverage: Dict[str, Any]
+    ):
+        """Parse STT-specific coverage fields from XML - only for internal planning items"""
+        planning_elt = news_coverage_elt.find(self.qname("planning"))
+        if planning_elt is None:
+            return
+
+        has_stt_fields = False
+
+        # Check for STT-specific namespaces and qcodes
+        for subj in planning_elt.findall(self.qname("subject")):
+            qcode = subj.get("qcode", "")
+            if any(
+                [
+                    qcode == "sttinternaltext",
+                    qcode == "sttentryinfo",
+                    qcode.startswith("sttworkstatus:"),
+                    qcode.startswith("sttphotoaware:"),
+                    qcode.startswith("sttimagetypename:"),
+                ]
+            ):
+                has_stt_fields = True
+                break
+
+        # Check for internal description fields
+        if not has_stt_fields:
+            for defn in planning_elt.findall(self.qname("definition")):
+                role = defn.get("role", "")
+                if role in ("sttdescription:imagetype", "sttdescription:imagetarget"):
+                    has_stt_fields = True
+                    break
+
+        if not has_stt_fields:
+            if (
+                planning_elt.find(".//{http://www.stt.fi/internal}workstartdate")
+                is not None
+            ):
+                has_stt_fields = True
+
+        if not has_stt_fields:
+            return
+
+        # Parse headline
+        headline_elt = planning_elt.find(self.qname("headline"))
+        if headline_elt is not None and headline_elt.text:
+            coverage.setdefault("planning", {})["headline"] = headline_elt.text.strip()
+
+        # Parse scheduled/due date
+        workstartdate_elt = planning_elt.find(
+            ".//{http://www.stt.fi/internal}workstartdate"
+        )
+        if (
+            workstartdate_elt is not None
+            and workstartdate_elt.text
+            and not coverage.get("planning", {}).get("scheduled")
```

Comment:
Looks like a Coverage should _always_ have a [scheduled date](https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/feed_parsers/superdesk_planning_xml.py#L232).

For STT's use case, I would assume if there is a `workstartdate` defined, we should use that and fallback to what is already populated from core parser.